Source/SourceKittenFramework/library_wrapper.swift

```diff
@@ -0,0 +1,145 @@
+//
+//  library_wrapper.swift
+//  sourcekitten
+//
+//  Created by Norio Nomura on 2/20/16.
+//  Copyright © 2016 SourceKitten. All rights reserved.
+//
+
+import Foundation
+
+struct DynamicLinkLibrary {
+    let path: String
+    let handle: UnsafeMutablePointer<Void>
+    
+    func loadSymbol<T>(symbol: String) -> T {
+        let sym = Darwin.dlsym(handle, symbol)
+        if sym == nil {
+            let errorString = String(UTF8String: dlerror())
+            fatalError("Finding symbol \(symbol) failed: \(errorString)")
+        }
+        return unsafeBitCast(sym, T.self)
+    }
+}
+
+let toolchainLoader = Loader(paths: [
+    xcodeDefaultToolchainOverride,
+    toolchainDir,
+    xcodeSelectPath?.toolchainDir,
+    /*
+    Followings are used when `xcode-select -p` points "Command Line Tools OS X for Xcode",
+    but Xcode.app exists.
+    */
+    applicationsDir?.xcodeDeveloperDir.toolchainDir,
+    applicationsDir?.xcodeBetaDeveloperDir.toolchainDir,
+    userApplicationsDir?.xcodeDeveloperDir.toolchainDir,
+    userApplicationsDir?.xcodeBetaDeveloperDir.toolchainDir,
+    ].flatMap { path in
+        if let fullPath = path?.usrLibDir where fullPath.isFile {
+            return fullPath
+        }
+        return nil
+    })
+
+struct Loader {
+    let paths: [String]
+
+    func load(path: String) -> DynamicLinkLibrary {
+        let fullPaths = paths.map { $0.stringByAppendingPathComponent(path) }.filter { $0.isFile }
+
+        // try all fullPaths that contains target file
+        for fullPath in fullPaths {
```

Comment:
`paths` contains search paths of loader.
Maybe I should rename that property to `searchPaths`.